include/graphqlservice/GraphQLService.h

```diff
@@ -655,17 +643,18 @@ struct ModifiedResult
 
 	// Peel off list modifiers.
 	template <TypeModifier Modifier, TypeModifier... Other>
-	static typename std::enable_if_t<TypeModifier::List == Modifier, std::future<response::Value>>
+	static typename std::enable_if_t<TypeModifier::List == Modifier, std::future<ResolverResult>>
 	convert(typename ResultTraits<Type, Modifier, Other...>::future_type result,
 		ResolverParams&& params)
 	{
 		return std::async(
 			params.launch,
 			[](auto&& wrappedFuture, ResolverParams&& wrappedParams) {
 				auto wrappedResult = wrappedFuture.get();
-				std::queue<std::future<response::Value>> children;
+				std::vector<std::future<ResolverResult>> children;
 
-				wrappedParams.errorPath.push(size_t { 0 });
+				children.reserve(wrappedResult.size());
+				wrappedParams.errorPath.push_back(size_t { 0 });
```

Comment:
this part you should avoid, you modify the list, then force the copy of the modified (`ResolverParams(wrappedParams)`), which is confusing and code become uglier.

Take a look at my PR, there I just keep a reference to the previous `SelectionSetParams`, then `errorPath` is not even a list/queue/vector anymore and it becomes copy-free, each `SelectionSetParams` is a node in an implicit linked list. You just pay the price on errors, which we expect to be not the common case.

Also the code becomes simpler, no need to push, set, reset... the index

Reply:
I have been experimenting with a version of this along with the mempool in https://github.com/wravery/cppgraphqlgen/tree/parser-mempool. The thing I don't particularly like about putting the link in the `SelectionSetParams` is that there are places in client code (e.g. the EdgeContraints template in TodayMock) which use move semantics on just the `errorPath`, and then it outlives the `SelectionSetParams`. So if it's relying on a reference to the parent `SelectionSetParams` struct, or even a reference to another `field_path`, the memory might have been freed by the time it tries to dereference the moved copy. It's not trivially relocatable from a single node, it needs to do a deep copy or otherwise keep a separate allocation alive for the parent.

To get around this I tried wrapping the segments in a `std::shared_ptr` and building a linked list from each segment to its parent. That was a little slower, and it used more memory. Most recently I tried allocating them on the stack and only copying the parent into a `shared_ptr` if a child is allocated on the stack. That seems slightly better, but it's still hitting `std::shared_ptr` quite a bit.

Reply:
I think we're paying a huge price to cope with this `EdgeConstraints` usage...

Actually, why do we need this to be handled by that class? Shouldn't the class just throw? I recall some parts of the code detected there was no path/location information and would properly enhance the errors before copying them.

`shared_ptr` doesn't seem like the way to go, in particular if we keep changing the `errorPath` as currently done by the list converter (something I'd also avoid).

Reply:
I've been worried about breaking backwards compatibility, figuring that existing external code might have implemented this like the sample. But I went back and checked the history for the sample, and I already modified it to add this reference in a minor version update (https://github.com/microsoft/cppgraphqlgen/releases/tag/v3.1.0). So maybe that's enough of a precedent to change this and drop the new code from EdgeConstraints. As you said, it should be able to just throw the exception without decoration and let the shared resolver code fill it in.

Reply:
It should noticeably break the build though if it's still used this way, since otherwise users will get a runtime error and crash.